SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -1331,35 +1339,29 @@ void USpatialActorChannel::ServerProcessOwnershipChange()
 	bool bUpdatedThisActor = false;
 
 	// Changing an Actor's owner can affect its NetConnection so we need to reevaluate this.
-	FString NewClientConnectionWorkerId = SpatialGDK::GetConnectionOwningWorkerId(Actor);
-	if (SavedConnectionOwningWorkerId != NewClientConnectionWorkerId)
+	check(NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::NET_OWNING_CLIENT_WORKER_COMPONENT_ID));
+	SpatialGDK::NetOwningClientWorker* CurrentNetOwningClientData = NetDriver->StaticComponentView->GetComponentData<SpatialGDK::NetOwningClientWorker>(EntityId);
+	const FString CurrentClientWorkerId = CurrentNetOwningClientData->WorkerId.IsSet() ? CurrentNetOwningClientData->WorkerId.GetValue() : FString();
+	const FString NewClientConnectionWorkerId = SpatialGDK::GetConnectionOwningWorkerId(Actor);
+	if (CurrentClientWorkerId != NewClientConnectionWorkerId)
 	{
 		// Update the NetOwningClientWorker component.
-		check(NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::NET_OWNING_CLIENT_WORKER_COMPONENT_ID));
-		SpatialGDK::NetOwningClientWorker* NetOwningClientWorkerData = NetDriver->StaticComponentView->GetComponentData<SpatialGDK::NetOwningClientWorker>(EntityId);
-		NetOwningClientWorkerData->WorkerId = NewClientConnectionWorkerId;
-		FWorkerComponentUpdate Update = NetOwningClientWorkerData->CreateNetOwningClientWorkerUpdate();
+		CurrentNetOwningClientData->SetWorkerId(NewClientConnectionWorkerId);
+		CurrentNetOwningClientData->SetPartitionId(SpatialGDK::GetConnectionOwningPartitionId(Actor));
+		FWorkerComponentUpdate Update = CurrentNetOwningClientData->CreateNetOwningClientWorkerUpdate();
 		NetDriver->Connection->SendComponentUpdate(EntityId, &Update);
 
-		// Update the EntityACL component (if authoritative).
-		if (NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::ENTITY_ACL_COMPONENT_ID))
-		{
-			Sender->UpdateClientAuthoritativeComponentAclEntries(EntityId, NewClientConnectionWorkerId);
-		}
-
-		SavedConnectionOwningWorkerId = NewClientConnectionWorkerId;
```

Comment:
Updating NetOwningClientData will initiate authority updates, so doing this here is redundant.